### PR TITLE
refs #49207 Default spaces

### PR DESCRIPTION
**Pull request by victor-aston**

### CODE REVIEW

File: modules/living_spaces_default/living_spaces_default.module [hunk 1]
```diff
@@ -96,6 +98,26 @@ function living_spaces_default_group_operations_alter(array &$operations, GroupI
...
 
+/**
+ * Implements hook_cron().
+ */
+function living_spaces_default_cron() {
+  $query = \Drupal::database()->select('group_relationship_field_data', 'gr');
+  $query->fields('gr', ['id']);
+  $query->distinct();
+  $query->join('groups_field_data', 'g', 'g.id = gr.gid');
+  $query->condition('g.is_default', 1);
```
Comment (andrew-sereda) [inline modules/living_spaces_default/living_spaces_default.module]: Add some range to not load/delete all relationships at once

**Review by andrew-sereda — CHANGES_REQUESTED** [review]

File: modules/living_spaces_default/living_spaces_default.module (hunk 1 above)
Comment (victor-aston) [reply modules/living_spaces_default/living_spaces_default.module]: added range.

File: modules/living_spaces_default/src/LivingSpacesDefaultRelationshipStorage.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Defines the storage handler class for relationship entities.
+ */
+class LivingSpacesDefaultRelationshipStorage extends GroupRelationshipStorage {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(EntityInterface $entity) {
+    if ($entity->getGroup()->get('is_default')->getString()) {
```
Comment (andrew-sereda) [inline modules/living_spaces_default/src/LivingSpacesDefaultRelationshipStorage.php]: This should only happen to group_membership.  
\Drupal\living_spaces_default\Access\LivingSpacesDefaultAccessCheck::access already handles access to adding content to space templates. If user is allowed to add content to space tempate, this code will remove relationship and related entity will just remain hanging

**Review by andrew-sereda — CHANGES_REQUESTED** [review]

File: modules/living_spaces_default/src/LivingSpacesDefaultRelationshipStorage.php (hunk 2 above)
Comment (victor-aston) [reply modules/living_spaces_default/src/LivingSpacesDefaultRelationshipStorage.php]: done

**Review by andrew-sereda — APPROVED** [review]

**Pull request merged**